app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,64 @@
+class AlbumsController < ApplicationController
+  def get_album
+    id = params[:id]
+    @album = Album.find(id)
+  end
+
+  def index
+    @albums = Album.all
+    @albums_sort = @albums.sort_by{ |album| album[:rank] }.reverse
+  end
+
+  def show
+    get_album
+  end
+
+  def new
+    @album = Album.new
+    @title = "Add New album"
+    @action = "create"
+  end
+
+  def create
+    @album = Album.create(album_params[:album])
+    @album.update(:rank => 0)
+    if @album.save
+      redirect_to "/albums"
+    else
+      render "/albums/new"
+    end
+
+  end
+
+  def edit
+    @album = Album.find(params[:id])
+    @title = "Edit #{@album.name}"
```

Comment:
Since you're changing the strings displayed in the form partial here and in the new action above, you could probably do away with the one-line new & edit views and have both render the form instead. But great use of partials to DRY up your code!